x-pack/legacy/plugins/siem/public/components/timeline/query_bar/index.tsx

```diff
@@ -157,15 +157,19 @@ export const QueryBarTimeline = memo<QueryBarTimelineComponentProps>(
       let isSubscribed = true;
       async function setSavedQueryByServices() {
         if (savedQueryId != null && savedQueryServices != null) {
-          const mySavedQuery = await savedQueryServices.getSavedQuery(savedQueryId);
-          if (isSubscribed) {
-            setSavedQuery({
-              ...mySavedQuery,
-              attributes: {
-                ...mySavedQuery.attributes,
-                filters: filters.filter(f => f.meta.controlledBy !== timelineFilterDropArea),
-              },
-            });
+          try {
+            const mySavedQuery = await savedQueryServices.getSavedQuery(savedQueryId);
+            if (isSubscribed && mySavedQuery != null) {
+              setSavedQuery({
+                ...mySavedQuery,
+                attributes: {
+                  ...mySavedQuery.attributes,
+                  filters: filters.filter(f => f.meta.controlledBy !== timelineFilterDropArea),
+                },
+              });
+            }
+          } catch (exc) {
+            setSavedQuery(null);
```

Comment:
Optional: I would add a comment to let others know when and why this exception is thrown? Also optional, I would print the stack trace on the console out as well so if it does happen we know to fix something.

I am usually careful of try catch do nothing blocks as later if we have a syntax error we will not find it until we get logic errors or test failures. With console log statements at least we will see them during test.

Is this try catch a result of us not having mocks for tests or is it a valid runtime throw? If it's due to issues with test mocks I would fix those. If it's a valid runttime error I would console log it at the very least.

Reply:
I agree - definitely worth updating the tests to show when / how the catch is triggered. A comment right in the code is a good idea too 👍